token/js/src/extensions/extensionType.ts

```diff
@@ -8,29 +8,34 @@ import { DEFAULT_ACCOUNT_STATE_SIZE } from './defaultAccountState/index.js';
 import { IMMUTABLE_OWNER_SIZE } from './immutableOwner.js';
 import { INTEREST_BEARING_MINT_CONFIG_STATE_SIZE } from './interestBearingMint/state.js';
 import { MEMO_TRANSFER_SIZE } from './memoTransfer/index.js';
+import { METADATA_POINTER_SIZE } from './metadataPointer/state.js';
 import { MINT_CLOSE_AUTHORITY_SIZE } from './mintCloseAuthority.js';
 import { NON_TRANSFERABLE_SIZE, NON_TRANSFERABLE_ACCOUNT_SIZE } from './nonTransferable.js';
 import { PERMANENT_DELEGATE_SIZE } from './permanentDelegate.js';
 import { TRANSFER_FEE_AMOUNT_SIZE, TRANSFER_FEE_CONFIG_SIZE } from './transferFee/index.js';
 import { TRANSFER_HOOK_ACCOUNT_SIZE, TRANSFER_HOOK_SIZE } from './transferHook/index.js';
 
 export enum ExtensionType {
-    Uninitialized,
-    TransferFeeConfig,
-    TransferFeeAmount,
-    MintCloseAuthority,
-    ConfidentialTransferMint,
-    ConfidentialTransferAccount,
-    DefaultAccountState,
-    ImmutableOwner,
-    MemoTransfer,
-    NonTransferable,
-    InterestBearingConfig,
-    CpiGuard,
-    PermanentDelegate,
-    NonTransferableAccount,
-    TransferHook,
-    TransferHookAccount,
+    Uninitialized = 0,
```

Comment:
There were some typescript errors if I created the unused `ConfidentialTransferFee`... instructions. Would need to define them in the below functions to work, and i'm not sure the approriate ones. 

The explicit definition is the same ones that Typescript automatically generates, so there is no change in the transplied javacript code.

Reply:
Which TypeScript errors? I would much rather just add them to the enum and do nothing else, so that we don't have to track these numbers.

Reply:
I think theres two parts to this discussion.


1. Want to clarify that the numbers are actually used and need to match the values in rust, so they do need to be tracked. For example the below function relies on the numbers to unpack the extension data. I prefer it explicitly defined, its a good enforcement that they actually have meaning, but will defer to you. If we remove the numbers, we just need to make sure the the order matches the rust implementation, so the automatic values given by TS matches the rust.
https://github.com/solana-labs/solana-program-library/blob/master/token/js/src/extensions/extensionType.ts#L189

1. The reason that unimplemented extensions like `ConfidentialTransferFee` are commented out, is that if they are defined it needs to be added to this function https://github.com/solana-labs/solana-program-library/blob/master/token/js/src/extensions/extensionType.ts#L130C33-L130C33. Theres no default case, so typescript will complain because the function can return  undefined. While its an easy fix to add a default cause to throw an error - i thought it would be better for whoever is implementing the extension to enable it and configure it correctly. Happy to add a default case if preferred.

Reply:
> 1. If we remove the numbers, we just need to make sure the the order matches the rust implementation

Yes, this is what we've been doing.

> 2\. While its an easy fix to add a default cause to throw an error

Yes do this! If the library doesn't support the extension yet, we just need to mark it as "unimplemented", which we can temporarily do with an error. Just so that it doesn't block progress on this PR.

Reply:
Okay, good that we are on the same page. I've gone ahead and removed it. 

However for 2 - it wasn't as easy as adding default error. This test case requires every extension to be either a Mint or Account extension, and i'm not confident which ones they should be.
https://github.com/solana-labs/solana-program-library/blob/master/token/js/test/unit/index.test.ts#L244

I removed all the numbering and only explicitly given a number to `MetadataPointer` as a work around. Take a look and hopefully its an acceptable compromise.